src/test/java/bio/terra/app/controller/DatasetsApiControllerTest.java

```diff
@@ -239,6 +243,51 @@ void testUpdateSnapshotBuilderSettings() throws Exception {
     verify(snapshotBuilderService).updateSnapshotBuilderSettings(DATASET_ID, SETTINGS);
   }
 
+  @Test
+  void testCreateSnapshotRequest() throws Exception {
+    mockValidators();
+    SnapshotBuilderAccessRequest expected =
+        new SnapshotBuilderAccessRequest()
```

Comment:
Another approach here would be to write the object as JSON. If it's less code I would do it that way instead.